salt/utils/dictupdate.py

```diff
@@ -17,7 +17,7 @@
 log = logging.getLogger(__name__)
 
 
-def update(dest, upd, recursive_update=True, merge_lists=False):
+def update(dest, upd, recursive_update=True, merge_lists=True):
```

Comment:
Given that commit 4af5b5c33f8b082797eb5827f59c4681ff9fba03 added this option and set the default of merge_lists=False, a non-existent option until that commit, setting it to True by default would in essence, revert things back to the original functionality where lists where merged unless otherwise so chosen via a new configuration option of pillar_merge_lists in the master config.  Leaving this at False is much more likely to cause regressions imo, than leaving only some changes set back to True or in other words, the behavior prior to commit 4af5b5c33f8b082797eb5827f59c4681ff9fba03.